### ISSUE TITLE
[Bug]: Library Fails on Firefox

### ISSUE THREAD

**Issue by DHJayasinghe** [damienbod/angular-auth-oidc-client#1487]
**What Version of the library are you using?**
13.1.0

**Describe the bug**
When I'm initiating my Angular app from Firefox browser i'm receiving an console error: DataError: **Data provided to an operation does not meet requirements** from **token-validation.service.ts** **validateSignatureIdToken()** method. With the App works fine on other browsers Chrome, Edge, And Brave browsers.

After building the library locally and debug, saw this [commit](https://github.com/damienbod/angular-auth-oidc-client/commit/8f10fee4dc58748ba2279911e94fa98f2b97f02e) which set's the **JsonWebKey.alg** property to null (when the browser agent is **Firefox** and the Signing Key using is **EC**), which causes the issue. Removed it and build the library locally, and the firefox failure gone.

Not sure what's the purpose of that condition which requires setting **JsonWebKey.alg** to **null** for that specific scenario.

**To Reproduce**
1. Using IdentityProvider which uses **ES256** (Eliptic Curve) Signing Algorithm for Tokens.
2. Using Firefox Browser to open the client app which uses angular-auth-oidc-client library.

**Expected behavior**
Libray should works on Firefox browser.

**Screenshots**
![image](https://user-images.githubusercontent.com/26274468/180700441-d56273d6-c298-4845-aed7-a6f0e4b71e2b.png)
![image](https://user-images.githubusercontent.com/26274468/180701243-c997cda4-cfc9-41db-aaea-5a95afacee43.png)

**Desktop (please complete the following information):**

- OS: Windows 10 Pro
- Browser Firefox
- Version 102.0.1

**Additional context**

Configuration:
```
{
        configId: environment.idp.clientId,
        authority: environment.idp.authority,
        redirectUrl: window.location.origin + '/sign-in',
        postLogoutRedirectUri: window.location.origin + '/sign-out',
        clientId: environment.idp.clientId,
        scope: environment.idp.scope,
        responseType: 'code',
        silentRenew: true,
        silentRenewTimeoutInSeconds: 60 * 1440, // 1 day
        useRefreshToken: true,
        enableIdTokenExpiredValidationInRenew: false,
        renewTimeBeforeTokenExpiresInSeconds: 150,
        tokenRefreshInSeconds: 30,
        logLevel: LogLevel.Debug,
        autoUserInfo: true,
        secureRoutes: [environment.baseUrl],
      }
```

Debug output - Firefox

```
[DEBUG] *redacted client_id* - Working with config '*redacted client_id*' using https://localhost:8000 angular-auth-oidc-client.mjs:160:20
[DEBUG] *redacted client_id* - currentUrl to check auth with:  http://localhost:4200/sign-in?code=D5A9F830FCF8401B598B76355…MVdkPFm0sc1fGKhlVNjbdSFdBJU.D9249FD8B38DD865E41274030B30B615 angular-auth-oidc-client.mjs:157:20
[DEBUG] *redacted client_id* - running validation for callback http://localhost:4200/sign-in?code=D5A9F830FCF8401B598B76355…MVdkPFm0sc1fGKhlVNjbdSFdBJU.D9249FD8B38DD865E41274030B30B615 angular-auth-oidc-client.mjs:157:20
[DEBUG] *redacted client_id* - AuthResult '{
  "id_token": "*redacted id_token*",
  "access_token": "*redacted access_token*",
  "expires_in": 1800,
  "token_type": "Bearer",
  "refresh_token": "*redacted refresh_token*",
  "scope": "*redacted scope*",
  "state": "2f1ab4ba969b32081ca0ec0e823461f82a3FmuTYX",
  "session_state": "GbDG06XwOoHEn5-CMVdkPFm0sc1fGKhlVNjbdSFdBJU.D9249FD8B38DD865E41274030B30B615"
}'.
      AuthCallback created, begin token validation angular-auth-oidc-client.mjs:160:20
[DEBUG] *redacted client_id* - Getting signinkeys from  https://localhost:8000/.well-known/openid-configuration/jwks angular-auth-oidc-client.mjs:157:20
[ERROR] *redacted client_id* - DataError: Data provided to an operation does not meet requirements angular-auth-oidc-client.mjs:196:20
ERROR Error: Uncaught (in promise): Error: Cannot match any routes. URL Segment: 'unauthorized'
noMatchError@http://localhost:4200/vendor.js:141905:12
apply/<@http://localhost:4200/vendor.js:141883:20
catchError/</innerSub<@http://localhost:4200/vendor.js:2808:94
OperatorSubscriber/this._error<@http://localhost:4200/vendor.js:2671:28
error@http://localhost:4200/vendor.js:1625:18
_error@http://localhost:4200/vendor.js:1649:30
error@http://localhost:4200/vendor.js:1625:18
_error@http://localhost:4200/vendor.js:1649:30
error@http://localhost:4200/vendor.js:1625:18
_error@http://localhost:4200/vendor.js:1649:30
error@http://localhost:4200/vendor.js:1625:18
_error@http://localhost:4200/vendor.js:1649:30
error@http://localhost:4200/vendor.js:1625:18
_error@http://localhost:4200/vendor.js:1649:30
error@http://localhost:4200/vendor.js:1625:18
_error@http://localhost:4200/vendor.js:1649:30
error@http://localhost:4200/vendor.js:1625:18
_error@http://localhost:4200/vendor.js:1649:30
error@http://localhost:4200/vendor.js:1625:18
init@http://localhost:4200/vendor.js:2575:45
_trySubscribe@http://localhost:4200/vendor.js:1258:25
subscribe/<@http://localhost:4200/vendor.js:1252:30
errorContext@http://localhost:4200/vendor.js:5350:9
subscribe@http://localhost:4200/vendor.js:1243:73
catchError/<@http://localhost:4200/vendor.js:2821:27
operate/</<@http://localhost:4200/vendor.js:5654:28
subscribe/<@http://localhost:4200/vendor.js:1247:30
errorContext@http://localhost:4200/vendor.js:5350:9
subscribe@http://localhost:4200/vendor.js:1243:73
map/<@http://localhost:4200/vendor.js:3221:16
```

**Comment by anwalkers** [comment]
I am having a similar issue with code flow on v13. My cleansed log. It is logging the user out when it goes to check the session. Time stamp 11:30:16.081 is where our application appears, then the user is immediately logged out.

Bad log from Firefox browser
```
11:30:06.542 Navigated to http://localhost:3000/
11:30:08.839 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - Working with config '0-6c4b2dea-2fff-477a-b641-222a7860fb25' using https://identity-server-url-removed vendor.js:393802:15
11:30:08.839 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - currentUrl to check auth with:  http://localhost:3000/ vendor.js:393800:15
11:30:08.839 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - checkAuth completed - firing events now. isAuthenticated: false vendor.js:393802:15
11:30:08.848 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - Working with config '0-6c4b2dea-2fff-477a-b641-222a7860fb25' using https://identity-server-url-removed vendor.js:393802:15
11:30:08.849 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - currentUrl to check auth with:  http://localhost:3000/index.html vendor.js:393800:15
11:30:08.849 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - checkAuth completed - firing events now. isAuthenticated: false vendor.js:393802:15
11:30:08.849 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - BEGIN Authorize OIDC Flow, no auth data vendor.js:393802:15
11:30:08.850 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - Nonce created. nonce:9cd24cdbf784986d2583e671c981bf4af4rvGWiTi vendor.js:393802:15
11:30:08.850 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - Authorize created. adding myautostate: 6eec74bd40d36bf5d588b3ab6b735deed4ekCfmJz vendor.js:393802:15
11:30:08.873 Navigated to https://identity-server-url-removed/connect/authorize?client_id=6c4b2dea-2fff-477a-b641-222a7860fb25&redirect_uri=http%3A%2F%2Flocalhost%3A3000%2Findex.html&response_type=id_token%20token&scope=openid%20roles%20groups%20profile%20email%20authorizations&nonce=9cd24cdbf784986d2583e671c981bf4af4rvGWiTi&state=6eec74bd40d36bf5d588b3ab6b735deed4ekCfmJz
11:30:13.465 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - Working with config '0-6c4b2dea-2fff-477a-b641-222a7860fb25' using https://identity-server-url-removed angular-auth-oidc-client.mjs:160:20
11:30:13.465 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - currentUrl to check auth with:  http://localhost:3000/index.html#id_token=eyJhbGciOiJSUzI1Ni…tOo9c2nlyDXyTA12YbQjXxGbhyg.FF50BD14928B93E2C57713DEB1FF6DF2 angular-auth-oidc-client.mjs:157:20
11:30:13.465 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - BEGIN callback, no auth data angular-auth-oidc-client.mjs:160:20
11:30:13.467 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - AuthResult '{
  "id_token": "ID_TOKEN_REMOVED",
  "access_token": "0C7469ACD16683B1F45F5495475E81E110C371A12F16AC2643566D0007CCD2F1",
  "token_type": "Bearer",
  "expires_in": "3600",
  "scope": "openid%20roles%20groups%20profile%20email",
  "state": "6eec74bd40d36bf5d588b3ab6b735deed4ekCfmJz",
  "session_state": "V_zLu1lbDnAzodTXtOo9c2nlyDXyTA12YbQjXxGbhyg.FF50BD14928B93E2C57713DEB1FF6DF2"
}'.
      AuthCallback created, begin token validation angular-auth-oidc-client.mjs:160:20
11:30:13.468 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - Getting signinkeys from  https://identity-server-url-removed/.well-known/openid-configuration/jwks angular-auth-oidc-client.mjs:157:20
11:30:13.752 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - validate id token iat max offset 2000 < 120000 angular-auth-oidc-client.mjs:160:20
11:30:13.753 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - Has idToken expired: false --> expires in 4:58 , 11:35:11 AM > 11:30:13 AM angular-auth-oidc-client.mjs:160:20
11:30:13.754 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - at_hash from the server:To1wzei-unjidXxYjR1eDw angular-auth-oidc-client.mjs:160:20
11:30:13.755 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - at_hash client validation not decoded:To1wzei-unjidXxYjR1eDw angular-auth-oidc-client.mjs:160:20
11:30:13.756 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - authCallback token(s) validated, continue angular-auth-oidc-client.mjs:160:20
11:30:13.756 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - storing the accessToken '0C7469ACD16683B1F45F5495475E81E110C371A12F16AC2643566D0007CCD2F1' angular-auth-oidc-client.mjs:160:20
11:30:14.744 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - Received user data:  
Object { sub: "eee3143f38c14e97b24e7f49f9891d56", given_name: "Given", family_name: "Family", name: "Family, Given", email: "Given.Family@mail.com", preferred_username: "username@domain.com" }
angular-auth-oidc-client.mjs:157:20
11:30:14.745 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - accessToken:  0C7469ACD16683B1F45F5495475E81E110C371A12F16AC2643566D0007CCD2F1 angular-auth-oidc-client.mjs:157:20
11:30:14.747 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - Has idToken expired: false --> expires in 4:27 , 11:35:11 AM > 11:30:44 AM angular-auth-oidc-client.mjs:160:20
11:30:14.747 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - Has accessToken expired: false --> expires in 59:29 , 12:30:13 PM > 11:30:44 AM angular-auth-oidc-client.mjs:160:20
11:30:14.747 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - persisted idToken and accessToken are valid angular-auth-oidc-client.mjs:160:20
11:30:14.747 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - [object HTMLIFrameElement] 2 angular-auth-oidc-client.mjs:160:20
11:30:14.769 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - checkAuth completed - firing events now. isAuthenticated: true angular-auth-oidc-client.mjs:160:20
11:30:14.826 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - Working with config '0-6c4b2dea-2fff-477a-b641-222a7860fb25' using https://identity-server-url-removed angular-auth-oidc-client.mjs:160:20
11:30:14.826 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - currentUrl to check auth with:  http://localhost:3000/index.html angular-auth-oidc-client.mjs:157:20
11:30:14.827 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - Has idToken expired: false --> expires in 4:27 , 11:35:11 AM > 11:30:44 AM angular-auth-oidc-client.mjs:160:20
11:30:14.827 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - Has accessToken expired: false --> expires in 59:29 , 12:30:13 PM > 11:30:44 AM angular-auth-oidc-client.mjs:160:20
11:30:14.827 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - persisted idToken and accessToken are valid angular-auth-oidc-client.mjs:160:20
11:30:14.830 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - checkAuth completed - firing events now. isAuthenticated: true angular-auth-oidc-client.mjs:160:20
11:30:15.280 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - CheckSession - clientId : '6c4b2dea-2fff-477a-b641-222a7860fb25' - existingIframe: '[object HTMLIFrameElement]' angular-auth-oidc-client.mjs:160:20
11:30:15.342 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - CheckSession - [object MessageEvent] from check session messageHandler angular-auth-oidc-client.mjs:160:20
11:30:16.077 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - Working with config '0-6c4b2dea-2fff-477a-b641-222a7860fb25' using https://identity-server-url-removed angular-auth-oidc-client.mjs:160:20
11:30:16.077 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - currentUrl to check auth with:  http://localhost:3000/index.html angular-auth-oidc-client.mjs:157:20
11:30:16.078 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - Has idToken expired: false --> expires in 4:25 , 11:35:11 AM > 11:30:46 AM angular-auth-oidc-client.mjs:160:20
11:30:16.078 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - Has accessToken expired: false --> expires in 59:27 , 12:30:13 PM > 11:30:46 AM angular-auth-oidc-client.mjs:160:20
11:30:16.078 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - persisted idToken and accessToken are valid angular-auth-oidc-client.mjs:160:20
11:30:16.081 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - checkAuth completed - firing events now. isAuthenticated: true angular-auth-oidc-client.mjs:160:20
11:30:16.834 Navigated to http://localhost:3000/logout.html
11:30:16.837 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - revocation endpoint post response:  null angular-auth-oidc-client.mjs:157:20
11:30:16.837 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - logoff, remove auth angular-auth-oidc-client.mjs:160:20
11:30:16.841 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - only local login cleaned up, server session has changed angular-auth-oidc-client.mjs:160:20
```

Here is the log from the Chrome browser. I have truncated it a bit, starting a few steps before where the Firefox browser logs the user out. The log out at the end was manually executed.

```
VM18:393802 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - persisted idToken and accessToken are valid
VM18:393802 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - checkAuth completed - firing events now. isAuthenticated: true
VM18:393802 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - CheckSession - clientId : '6c4b2dea-2fff-477a-b641-222a7860fb25' - existingIframe: '[object HTMLIFrameElement]'
VM18:393802 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - CheckSession - unchanged from check session messageHandler
VM18:393802 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - Working with config '0-6c4b2dea-2fff-477a-b641-222a7860fb25' using https://idenservertest2.itrontotaltest.com
VM18:393800 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - currentUrl to check auth with:  http://localhost:3000/index.html
VM18:393802 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - Has idToken expired: false --> expires in 4:24 , 11:55:03 AM > 11:50:39 AM
VM18:393802 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - Has accessToken expired: false --> expires in 59:26 , 12:50:05 PM > 11:50:39 AM
VM18:393802 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - persisted idToken and accessToken are valid
VM18:393802 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - checkAuth completed - firing events now. isAuthenticated: true
** THIS IS WHERE IT LOGS THE USER OFF IN FIREFOX ** --> VM18:393802 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - CheckSession - clientId : '6c4b2dea-2fff-477a-b641-222a7860fb25' - existingIframe: '[object HTMLIFrameElement]'
VM18:393802 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - CheckSession - unchanged from check session messageHandler
VM18:393802 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - Checking: silentRenewRunning: false - has idToken: true - has userData: true
VM18:393802 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - Has idToken expired: false --> expires in 4:23 , 11:55:03 AM > 11:50:40 AM
VM18:393802 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - Has accessToken expired: false --> expires in 59:25 , 12:50:05 PM > 11:50:40 AM
VM18:393802 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - silent renew, periodic check finished!
VM18:393802 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - CheckSession - clientId : '6c4b2dea-2fff-477a-b641-222a7860fb25' - existingIframe: '[object HTMLIFrameElement]'
VM18:393802 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - CheckSession - unchanged from check session messageHandler
VM18:393802 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - Checking: silentRenewRunning: false - has idToken: true - has userData: true
VM18:393802 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - Has idToken expired: false --> expires in 4:19 , 11:55:03 AM > 11:50:44 AM
VM18:393802 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - Has accessToken expired: false --> expires in 59:21 , 12:50:05 PM > 11:50:44 AM
VM18:393802 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - silent renew, periodic check finished!
VM18:393802 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - CheckSession - clientId : '6c4b2dea-2fff-477a-b641-222a7860fb25' - existingIframe: '[object HTMLIFrameElement]'
VM18:393802 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - CheckSession - unchanged from check session messageHandler
VM18:393802 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - Checking: silentRenewRunning: false - has idToken: true - has userData: true
VM18:393802 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - Has idToken expired: false --> expires in 4:15 , 11:55:03 AM > 11:50:48 AM
VM18:393802 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - Has accessToken expired: false --> expires in 59:17 , 12:50:05 PM > 11:50:48 AM
VM18:393802 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - silent renew, periodic check finished!
VM18:393802 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - CheckSession - clientId : '6c4b2dea-2fff-477a-b641-222a7860fb25' - existingIframe: '[object HTMLIFrameElement]'
VM18:393802 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - CheckSession - unchanged from check session messageHandler
VM18:393802 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - CheckSession - clientId : '6c4b2dea-2fff-477a-b641-222a7860fb25' - existingIframe: '[object HTMLIFrameElement]'
VM18:393802 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - CheckSession - unchanged from check session messageHandler
VM18:393800 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - revocation endpoint post response:  null
VM18:393802 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - logoff, remove auth 
VM18:393802 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - Checking: silentRenewRunning: false - has idToken: false - has userData: false
VM18:393802 [DEBUG] 0-6c4b2dea-2fff-477a-b641-222a7860fb25 - silent renew, periodic check finished!
```

**Comment by anwalkers** [comment]
@damienbod It is only an issue against a localhost without ```--ssl``` options. 

I spun up your project on the ```release_13_1_0``` branch locally and your example project runs fine. I tested it on ubuntu 20.04 running FireFox 150.0.3 and Windows running FireFox 150.0.3 and 160.0.3. 

FireFox works fine with our deployed environment. I will investigate adding the ```--ssl``` option to our local development environment and see if the issue persists.

**Comment by mmarrano** [comment]
> OP's original description
I am running into this same problem today

**What Version of the library are you using?**
15.0.0

**Desktop (please complete the following information):**
* OS: Windows 10 Pro
* Browser Firefox
* Version 107.0

**Additional context**
Configuration:
```
{
      authority: this.config.config?.oidc.authority,
      redirectUrl: home,
      postLogoutRedirectUri: home,
      clientId: this.config.config?.clientId,
      scope: 'openid',
      responseType: 'id_token token',
      silentRenew: true,
      silentRenewUrl: home,
      renewTimeBeforeTokenExpiresInSeconds: 30,
      startCheckSession: true,
      logLevel: LogLevel.None,
      customParamsAuthRequest: {
        auth_method: isExternal ? '-----' : '-----',
        realm: isExternal ? '-----' : '-----',
        ttl: 1200,
        identity_method: '-----'
      },
      unauthorizedRoute: '/-----/unauthorized',
      forbiddenRoute: '/-----/unauthorized'
}
```

I effectively followed exactly what @DHJayasinghe did, cloned this library, removed the code he referenced, and implemented to my parent app. I immediately recognized relief from this problem and my application now works on Firefox. My problem wasn't like @anwalkers's either, I was seeing this issue locally and in all my deployed environments. `--ssl` did not fix this issue locally for me. I would like to note, that my provider is using `kty` of `EC` for its keys so that could be in play there.

I couldn't find any information on why that code was implemented back in 13.1.0, but I know my 'workaround' for now is reverting it.
